src/methods.jl

```diff
@@ -77,3 +74,63 @@ julia> subtract_overscan(frame, (:, 4:5), dims = 2)
 * [`subtract_overscan!`](@ref)
 """
 subtract_overscan(frame::AbstractArray, idxs; dims = axes_min_length(idxs)) = subtract_overscan!(deepcopy(frame), idxs, dims = dims)
+
+
+"""
+    flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+
+In place version of [`flat_correct`](@ref)
+"""
+function flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+    size(frame) != size(flat_frame) && error("size of frame and flat frame are not same")
+    !(norm_value === nothing) && norm_value < 0 && error("norm_value must be greater than 0")
+
+    !(min_value === nothing) && @. flat_frame[flat_frame < min_value] = min_value
+
+    if !(norm_value === nothing)
+        flat_frame ./= norm_value
+    else
+        flat_frame ./= mean(flat_frame)
+    end
+
+    frame ./= flat_frame
+    return frame
+end
+
+
+"""
+    flat_correct(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+
+Correct `frame` for non-uniformity using the calibrated `flat_frame`.
+
+`min_value` is minimum value of `flat_frame`, an entry less than `min_value` is converted to `min_value` in `flat_frame`.
+`norm_value` is normalising constant for `flat_frame`, if set to `nothing` then `mean` of `flat_frame` is used for normalising.
```

Comment:
The grammar here is a little sharp- it isn't that smooth to read, try

"If `min_value` is provided, the `flat_frame` will be minimum-clipped to that value. By default, the `flat_frame` is normalized by its mean, but this can be changed by providing a custom `norm_value`. "